### ISSUE TITLE
Push.appCollection still empty

### ISSUE THREAD

**Issue by tltim** [raix/push#89]
I read the other issues on this topic and tried all of the alternatives but I still can not get this to work for GCM. 
I have:
-Enabled GCM API for android
-Created the Push.config.json file with the API key and project number

However when I try pushing with meteor shell I get:
I20150714-13:37:51.687(-7)? Push: Send message "hello" via query {}
I20150714-13:37:51.689(-7)? Push: Sent message "hello" to 0 ios apps 0 android apps
I20150714-13:37:51.689(-7)? Push, GUIDE: The "Push.appCollection" is empty - No clients have registred on the server yet...

Client.js:
Push.addListener('token', function(token) {
  			console.log('Token: ' + JSON.stringify(token));
  			alert('Token: ' + JSON.stringify(token));
		});

I do get a token on startup but I still don't get any notifications even after logging in.

**Comment by chrisco23** [comment]
yep I have battled this one for the past 8 hours to no avail.  Please let me know if anybody has any new ideas to debug this as my app is completely dependent on notifications.

**Comment by raix** [comment]
try setting the `--mobile-server` flag - the client might not be able to connect to the server

**Comment by tltim** [comment]
I've tried --mobile-server <server>:<port> and without the <port> many times and it still does not work. I've tried many of the solutions from the other issues but I still can not get it to work. JSON.stringify reveals that I receive a token but it still does not register the client.

**Comment by chrisco23** [comment]
I meant to update this earlier but I guess I only updated another thread.  It turned out to be something with my phone and I still would like to know what it was exactly, myself.  In googling things like "can't receive Google GCM" and the like, I came across an app which you can easily find on Google Play, called Push Notification Tester.  Now, I downloaded that app, ran it, and the notification worked.  I then sent a test notification with my app and this raix package, no code changes whatsoever, and it just worked.  What can have changed?  Note that before running this tester app, I received notifications fine from, eg. meetup.com and other apps.  I do keep things like Facebook or Twitter notifications turned off, and I wondered if some per-app setting of "Off" might have been affecting my own meteor-w-raix-push app somehow...  I pretty much spent an extremely long day debugging ghosts.

**Comment by tltim** [comment]
Hi Chris, I downloaded the app you mentioned and tried it and I get a notification from that app but I am still unable to receive notifications on my phone with the package using meteor shell.

**Comment by chrisco23** [comment]
Shoot.  I'm trying to remember now any other little snags I hit.  I believe config.push.json needs to be at root level.  Dbl-check your api settings.  My config looks like this:

`
{
  "gcm": {
    "apiKey": "foo",
    "projectNumber": 1234
  },
  "badge": true,
  "sound": true,
  "alert": true,
  "production": false
}
`

At one point someone in IRC suggested I add this (in /client), although I don't know that any of this is necessary:
`
Meteor.startup(function () {
  Push.allow({
    send: function(userId, notification) {
      // Allow all users to send to everybody - For test only!
      return true;
    }
  });

  Push.debug = true;

  /* do something with push token*/
  Push.addListener('token', function(token) {
    console.log('Token: ' + JSON.stringify(token));
  });

  Push.addListener('notification', function(notification) {
    console.log('notification: ' + JSON.stringify(notification))
  });

  Push.addListener('startup', function(notification){
    console.log('Routing Push: ' + JSON.stringify(notification));
  });
});
`

Lastly, I am mup-deploying to a DO server.  IIRC, I only ever got things to work when running with --mobile-server to my mup-deployed server, and I was able to do some debugging with "DEBUG=* mup logs" after the "mup deploy".

Hopefully that can give you something new to try.  It was a very long day for me but in the end the problems seemed to be on my end, nothing with the package itself.

**Comment by tltim** [comment]
I've also been deploying with digital ocean and running with that mobile server, did you do anything in DO?

**Comment by chrisco23** [comment]
Nothing special really.  I use nginx proxy.  I have had the firewall turned
off since some other thing I was debugging.  I haven't tried turning the
firewall back on and tuning that yet.

**Comment by chrisco23** [comment]
And yes I "mup deploy" to a minimal $5 DO server.  Everything else with
your app is working on DO--just not notifications yet?

**Comment by tltim** [comment]
yeah everything works fine it's only notifications it also may have something to do with accounts? Because I get the token before logging in but not after. But even before logging in the Push.appCollection.find().fetch() is still empty.

**Comment by chrisco23** [comment]
I can't think of what it could be with accounts.  I'm using only
accounts-password and ian:accounts-ui-boostrap-3.  What did you see when
you ran the "DEBUG=* mup logs"?
